From b998982652f47efe120043d102a034fe3b834a96 Mon Sep 17 00:00:00 2001
From: Ankit Agrawal <ankit.agarwal@intel.com>
Date: Thu, 6 Jul 2023 11:38:59 +0530
Subject: [PATCH] Multi-passenger feature support.

if dynamic routing is not enabled, then default assign
primary zone id.

Tracked-On: OAM-109879
Signed-off-by: Ankit Agrawal <ankit.agarwal@intel.com>
---
 .../settings/sound/VolumeSettingsPreferenceController.java | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/com/android/car/settings/sound/VolumeSettingsPreferenceController.java b/src/com/android/car/settings/sound/VolumeSettingsPreferenceController.java
index 839dcf16c..6341e5707 100644
--- a/src/com/android/car/settings/sound/VolumeSettingsPreferenceController.java
+++ b/src/com/android/car/settings/sound/VolumeSettingsPreferenceController.java
@@ -16,6 +16,7 @@
 
 package com.android.car.settings.sound;
 
+import static android.car.media.CarAudioManager.AUDIO_FEATURE_DYNAMIC_ROUTING;
 import static android.car.media.CarAudioManager.AUDIO_FEATURE_VOLUME_GROUP_MUTING;
 import static android.car.media.CarAudioManager.PRIMARY_AUDIO_ZONE;
 import static android.os.UserManager.DISALLOW_ADJUST_VOLUME;
@@ -71,7 +72,7 @@ public class VolumeSettingsPreferenceController extends PreferenceController<Pre
     private final VolumeSettingsRingtoneManager mRingtoneManager;
 
     private final Handler mUiHandler;
-    private int mZoneId;
+    private int mZoneId = PRIMARY_AUDIO_ZONE;
 
     @VisibleForTesting
     final CarAudioManager.CarVolumeCallback mVolumeChangeCallback =
@@ -118,7 +119,9 @@ public class VolumeSettingsPreferenceController extends PreferenceController<Pre
         if (mCarAudioManager != null) {
             try {
                 ApplicationInfo info = context.getPackageManager().getApplicationInfo(context.getPackageName(), 0);
-                mZoneId = mCarAudioManager.getZoneIdForUid(info.uid);
+                if (mCarAudioManager.isAudioFeatureEnabled(AUDIO_FEATURE_DYNAMIC_ROUTING)) {
+                    mZoneId = mCarAudioManager.getZoneIdForUid(info.uid);
+                }
             } catch (PackageManager.NameNotFoundException e) {
                 LOG.e("getZoneIdForUid Failed to find zoneId: " , e);
             }
-- 
2.17.1

